packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -137,6 +137,7 @@ export default function jsx({ config, logging }: AstroPluginJSXOptions): Plugin
         const { code: jsxCode } = await esbuild.transform(code, {
           loader: getEsbuildLoader(path.extname(id)) as esbuild.Loader,
           jsx: 'preserve',
+          sourcefile: id,
```

Comment:
Big fix in React/Preact/Solid: when we pass the filename to esbuild, it generates a much more helpful stacktrace